### PR TITLE
Fixes for windows regressions in classfile writing

### PR DESCRIPTION
**Pull request by retronym**
Fixes scala/scala-dev#638

### CODE REVIEW

File: test/files/run/t5717.scala [hunk 1]
```diff
@@ -20,7 +20,8 @@ object Test extends StoreReporterDirectTest {
     val List(i) = filteredInfos
     // for some reason, nio doesn't throw the same exception on windows and linux/mac
     val path = if(util.Properties.isWin)"\\a" else "/a"
-    val expected = s"error writing ${testOutput.path}/a/B.class: Can't create directory ${testOutput.path}${path}" +
+    import File.pathSeparator
+    val expected = s"error writing ${testOutput.path}${pathSeparator}a${pathSeparator}B.class: Can't create directory ${testOutput.path}${pathSeparator}/a" +
```
Comment (som-snytt) [inline test/files/run/t5717.scala]: There is a residual forward slash.

The other time-tested method is just to replaceAll backslashes with forward slash to normalize the text.

File: test/files/run/t5717.scala [hunk 2]
```diff
@@ -20,7 +20,8 @@ object Test extends StoreReporterDirectTest {
     val List(i) = filteredInfos
     // for some reason, nio doesn't throw the same exception on windows and linux/mac
     val path = if(util.Properties.isWin)"\\a" else "/a"
```
Comment (som-snytt) [inline test/files/run/t5717.scala]: Xlint would have said no longer used.

File: test/files/run/t5717.scala [hunk 3]
```diff
@@ -20,7 +20,8 @@ object Test extends StoreReporterDirectTest {
...
-    val expected = s"error writing ${testOutput.path}/a/B.class: Can't create directory ${testOutput.path}${path}" +
+    import File.pathSeparator
+    val expected = s"error writing ${testOutput.path}${pathSeparator}a${pathSeparator}B.class: Can't create directory ${testOutput.path}${pathSeparator}/a" +
       "; there is an existing (non-directory) file in its path"
     assert(i.msg == expected, i.msg)
```
Comment (som-snytt) [inline test/files/run/t5717.scala]: Why we need assertEquals, or even better, `specty` power assert as the fundamental building block.

Comment (som-snytt) [reply test/files/run/t5717.scala]: Also thanks to you and Lukas for not complaining about this test from 2013. For some reason, committed on my birthday. I guess my gift to myself was to try something in Scala.

File: src/compiler/scala/tools/nsc/backend/jvm/ClassfileWriters.scala [hunk 4]
```diff
@@ -52,8 +52,8 @@ abstract class ClassfileWriters {
      */
     def close(): Unit
 
-    protected def classRelativePath(className: InternalName, suffix: String = ".class"): Path =
-      Paths.get(className.replace('.', '/') + suffix)
+    protected def classRelativePath(className: InternalName, suffix: String = ".class"): String =
+      className.replace('.', '/') + suffix
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/backend/jvm/ClassfileWriters.scala]: `classRelativeString` :)

File: test/files/run/t5717.scala (hunk 1 above)
Comment (som-snytt) [reply test/files/run/t5717.scala]: That was me recommending test string techniques to retronym.

**Comment by SethTisue** [comment]
green run: https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-windows/1431/ :+1:

**Comment by som-snytt** [comment]
Yay. I felt bad that some poor test I quote-unquote contributed in 2013 cost folks some time, over time.

**Comment by SethTisue** [comment]
0.001% of our gratitude for your contributions is CANCELED!

**Comment by som-snytt** [comment]
For some reason, I didn't remember this for https://github.com/scala/scala/pull/8539

**Pull request merged**